src/runtime/tests/particle-interface-loading-test.ts

```diff
@@ -293,4 +293,75 @@ describe('particle interface loading', () => {
     const fooHandle2 = await handleNGFor('crdt-key', varStorageProxy2, arc2.idGenerator, null, true, true, 'varHandle') as SingletonHandle<Entity>;
     assert.deepStrictEqual(await fooHandle2.fetch(), new fooClass({value: 'Not created!'}));
   });
+
+  it('onReady runs when handles are first synced', async () => {
```

Comment:
An idea test would be performed at a layer one level down, i.e. where we could sequentially make handles sync and see that onReady() is not called until we sync the last handle. The test at the level of instantiating a recipe cannot test that. Without looking through the codebase myself I cannot point you to a similar test, but maybe you could explore whether there are unit tests for the particle.ts that we could replicate?

Reply:
Added many more tests that I think ensure everything happens at the right time now.